presto-postgresql/src/main/java/com/facebook/presto/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -81,13 +89,35 @@ protected ResultSet getTables(Connection connection, String schemaName, String t
                 new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"});
     }
 
+    public static ReadMapping jsonReadMapping()
+    {
+        return sliceReadMapping(JSON, (resultSet, columnIndex) -> utf8Slice(resultSet.getString(columnIndex)));
+    }
+
     @Override
     protected String toSqlType(Type type)
     {
         if (VARBINARY.equals(type)) {
             return "bytea";
         }
+        else if (JSON.equals(type)) {
+            return "jsonb";
+        }
 
         return super.toSqlType(type);
     }
+
+    @Override
+    public Optional<ReadMapping> toPrestoType(ConnectorSession session, JdbcTypeHandle typeHandle)
+    {
+        if (typeHandle.getJdbcType() == Types.OTHER) {
+            switch (typeHandle.getJdbcTypeName()) {
+                case "jsonb":
+                case "json":
+                    return Optional.of(jsonReadMapping());
+            }
```

Comment:
Will predicate-pushdown work correctly with json? We definitely need some tests for that.

Reply:
Can you please elaborate about this and add some references to similar tests ?